Source/missiles.cpp

```diff
@@ -189,33 +189,9 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, missile_id t
 {
 	auto &monster = Monsters[m];
 
-	bool resist = false;
-	if (monster.mtalkmsg != TEXT_NONE
-	    || monster._mhitpoints >> 6 <= 0
-	    || (t == MIS_HBOLT && monster.MType->mtype != MT_DIABLO && monster.MData->mMonstClass != MonsterClass::Undead)) {
+	bool resist = monster.IsResistant(t);

```

Comment:
This should also be moved to the first use, which appear to be line 245

Reply:
ok, btw, who shall resolve conversation, you or me?

Reply:
Since I brought it up it would probably be me, unless you feel very sure that you have addressed it in a way that I would accept.